tests/ftests/089-cgset-recursive_flag.py

```diff
@@ -0,0 +1,143 @@
+    Cgroup.get_and_validate(config, GRANDCHILD, SETTING_V1, VALUE_V1)
+
+    Cgroup.set(config, cgname=PARENT, setting=SETTING_V1,
+               value=DEFAULT_VALUE_V1, recursive=False)
+
+    Cgroup.get_and_validate(config, PARENT, SETTING_V1, DEFAULT_VALUE_V1)
+    Cgroup.get_and_validate(config, CHILD, SETTING_V1,  VALUE_V1)
+    Cgroup.get_and_validate(config, GRANDCHILD, SETTING_V1, VALUE_V1)
+
+    return consts.TEST_PASSED, None
+
+
+def test_cgroupv2(config):
+    result = consts.TEST_PASSED
+    cause = None
+
+    Cgroup.set(config, cgname=PARENT, setting=SETTING_V2_SUBTREE,
+               value=VALUE_V2_SUBTREE, recursive=True)
+
+    # checking if the controller is enabled in granchildcg, in turn will
+    # check its parent cgroup childcg's subtree_control file, if it's enabled
+    # in parent cgroup, it's also enabled in the grandparent too.
+    if not Cgroup.is_controller_enabled(config, GRANDCHILD, CONTROLLER):
+        result = consts.TEST_FAILED
+        cause = 'Controller {} is not enabled in the child cgroup'.format(CONTROLLER)
+
+    # check if 'cpuset' controller is enabled
+    if not Cgroup.is_controller_enabled(config, GRANDCHILD, 'cpuset'):
+        result = consts.TEST_FAILED
+        tmp_cause = ('Controller {} is not enabled in the child cgroup'
+                     ''.format('cpuset'))
+        cause = '\n'.join(filter(None, [cause, tmp_cause]))
+
+    # check if 'pids' controller is enabled
+    if Cgroup.is_controller_enabled(config, GRANDCHILD, 'pids'):
+        result = consts.TEST_FAILED
+        tmp_cause = ('Controller {} is enabled in the child cgroup'
+                     ''.format('pids'))
+        cause = '\n'.join(filter(None, [cause, tmp_cause]))
+
+    # for the grandchildcg read it's cgroup.subtree_control, is_controller_enabled
+    # will check its parent cgroup childcg's subtree_control
+    if Cgroup.get(config, None, GRANDCHILD, setting='cgroup.subtree_control',
+                  print_headers=False, values_only=True):
+        result = consts.TEST_FAILED
+        tmp_cause = 'Controller {} enabled in grandchild cgroup'.format(CONTROLLER)
+        cause = '\n'.join(filter(None, [cause, tmp_cause]))
+
+    Cgroup.set(config, cgname=PARENT, setting=SETTING_V2, value=VALUE_V2,
+               recursive=True)
+
+    Cgroup.get_and_validate(config, PARENT, SETTING_V2, VALUE_V2)
+    Cgroup.get_and_validate(config, CHILD, SETTING_V2,  VALUE_V2)
+    Cgroup.get_and_validate(config, GRANDCHILD, SETTING_V2, VALUE_V2)
+
+    Cgroup.set(config, cgname=PARENT, setting=SETTING_V2,
+               value=DEFAULT_VALUE_V2, recursive=False)
+
+    Cgroup.get_and_validate(config, PARENT, SETTING_V2, DEFAULT_VALUE_V2)
+    Cgroup.get_and_validate(config, CHILD, SETTING_V2, VALUE_V2)
+    Cgroup.get_and_validate(config, GRANDCHILD, SETTING_V2, VALUE_V2)
+
+    return result, cause
+
+
+def test(config):
+
+    if Cgroup.get_cgroup_mode(config) == Mode.CGROUP_MODE_UNIFIED:
```

Comment:
I hate to ask this, but again what about hybrid mode?  Especially a configuration where some controllers are v1 and some are v2.

Reply:
@drakenclimber Sure will add the support for the `hybrid` mode too.